### PR TITLE
[PC-1929] Initial commit: Conveyor and sorting AppNote

### PR DESCRIPTION
**Pull request by DavidsonMSD**
- Set up the initial project structure.
  - Added main project files.
  - Added assets.
  - Added PLC IDE code (including both Ladder and Sketch).

## What This PR Changes
- This PR adds the initial version of the conveyor and sorting application note.

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+int  cloudBlueSorted;
+
+// WiFi connection handler for connecting to Arduino Cloud
+WiFiConnectionHandler ArduinoIoTPreferredConnection(SSID, PASS);
+
+```
+**Explanation:**
+
+1. **Library Inclusions**:
+   - `#include <ArduinoIoTCloud.h>`: This library provides the necessary functions to connect and interact with the Arduino Cloud.
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:435]: Nested lists not supported

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+   - `void onStartButtonChange();` and `void onStopButtonChange();`: These are function declarations for handling changes in the start and stop buttons state when received from the cloud.
+
+4. **Cloud-Linked Variables**:
+   - `bool cloudStartButton;`, `bool cloudStopButton;`, `bool cloudRunningSystem;`: These boolean variables represent the state of the start/stop buttons and the running system status in the cloud.
+   - `int cloudRedSorted;`, `int cloudGreenSorted;`, `int cloudBlueSorted;`: These integer variables track the number of red, green, and blue items sorted and are synchronized with the cloud.
+
+5. **WiFi Connection Handler**:
+   - `WiFiConnectionHandler ArduinoIoTPreferredConnection(SSID, PASS);`: This object manages the WiFi connection to the Arduino Cloud using the provided SSID and password.
+
+### **Second Section: Setup Function**
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:451]:
```suggestion
### Second Section: Setup Function
```

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+  setDebugMessageLevel(2);
+  ArduinoCloud.printDebugInfo();
+}
+```
+**Explanation:**
+
+The `setup` function in Arduino programming is a crucial function that executes once when the Arduino board is powered on or reset. It’s responsible for initializing settings and configuring the system before the main program starts running. In the `setup` function, you typically establish communication (such as starting the serial monitor), configure pin modes, connect to networks, and initialize variables and libraries. This function ensures the Arduino is fully prepared for the `loop` function that follows, allowing for continuous and smooth operation. In this sketch, we are setting up:
+
+1. **Serial Communication Initialization**:
+   - `Serial.begin(9600);`: Initializes the serial communication at a baud rate of 9600, allowing you to monitor the device's output through the Serial Monitor.
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:482]:
```suggestion
   - `Serial.begin(9600);`: Initializes the serial communication at a baud rate of 9600 bps, allowing you to monitor the device's output through the Serial Monitor.
```

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+   - `initProperties();`: This function is defined later in the code and is responsible for linking the local variables with the Arduino Cloud properties.
+
+4. **Arduino Cloud Connection**:
+   - `ArduinoCloud.begin(ArduinoIoTPreferredConnection);`: This command connects the device to the Arduino Cloud using the specified WiFi credentials.
+
+5. **Debugging Information**:
+   - `setDebugMessageLevel(2);`: Sets the level of debug information. A higher number provides more detailed information about network and cloud connection states. The default is 0 (only errors), and the maximum is 4 (most detailed).
+   - `ArduinoCloud.printDebugInfo();`: Prints debug information about the cloud connection to the Serial Monitor.
+
+**Third Section: Loop Function**
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:497]:
```suggestion
### Third Section: Loop Function
```

Comment (mcmchris) [reply content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:497]: For title consistency

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+5. **Red Items Sorted Count**:
+   - `cloudRedSorted = PLCOut.out_counterRed;`: Synchronizes the variable in the PLC with the variable in the Arduino Cloud.
+
+6. **Green Items Sorted Count**:
+   - `cloudGreenSorted = PLCOut.out_counterGreen;`: Synchronizes the variable in the PLC with the variable in the Arduino Cloud.
+
+7. **Blue Items Sorted Count**:
+   - `cloudBlueSorted = PLCOut.out_counterBlue;`: Synchronizes the variable in the PLC with the variable in the Arduino Cloud.
+
+**Fourth Section:  InitProperties and Callbacks**
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:539]:
```suggestion
### Fourth Section:  InitProperties and Callbacks
```

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+
+### Arduino Cloud Dashboard
+
+The Arduino Cloud provides a powerful platform for monitoring and controlling the conveyor and sorting system in real time. With this setup, you can remotely oversee the entire process and interact with key components of the system as they happen. The system tracks the total number of items sorted by color, red, green, and blue, allowing you to monitor how many of each have been processed.
+
+In addition to tracking the sorted items, the Arduino Cloud also allows you to control the start and stop of the system. You can initiate or stop the system directly from the cloud interface, giving you full control over the operation from anywhere with an internet connection, all in real time.
+
+This real-time integration ensures that you can both observe the performance of your sorting system and take direct action when needed, making it a versatile and effective tool for managing your operations instantly and efficiently.
+
+![Arduino Cloud Dashboard](assets/conveyor-dashboard.gif)
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:594]: Let's give you access to an enterprise Arduino Cloud plan so you can add more widgets and customize the dashboard.

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 7]
```diff
@@ -0,0 +1,621 @@
...
+| **Arduino_DebugUtils**         | 1.4.0   | Debugging module with different debug levels, timestamps and printf-style output. This class provides functionality useful for debugging sketches via printf-style statements. |
+| **Arduino_Portenta_OTA**       | 1.2.1   | Firmware update for the Portenta H7. This library allows performing a firmware update on the Arduino Portenta H7. The firmware can be stored in various different locations such as within the microcontroller's flash, on an external SD card or on the QSPI flash chip. |
+| **Arduino_SecureElement**      | 0.1.2   | This library allows usage of Arduino boards secure elements in a common and unified way. |
+
+These libraries are indexed. They are certified to ensure optimal performance and reliability. Using them will accelerate your development process and strengthen the durability of your applications, especially in industrial environments. By incorporating these libraries with the specified versions, developers can efficiently establish and maintain secure and reliable communication between Opta and the Arduino Cloud, enabling robust control, monitoring, and remote updates.
+
+### Arduino Sketch Code
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:313]: Would be nice to have a screenshot of the PLC IDE so reader can mentally locate in the software where the code will be written.

File: content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md [hunk 8]
```diff
@@ -0,0 +1,621 @@
...
+
+7. **System Stop**
+   - **Action:** Press the stop button.
+   - **Description:** The system is stoped, stopping all operations and deactivating the conveyors and actuator.
+
+8. **Remote Monitoring and Control**
+   - **Action:** Monitor and control the system via the Arduino Cloud.
+   - **Description:**  At any time, you can check the operating status and control the system remotely using the Arduino Cloud interface
+
+## Opta Ladder Code
```
Comment (mcmchris) [inline content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md:149]: Would be nice to have a screenshot of the PLC IDE so reader can mentally locate in the software where the Ladder diagram will be written.

**Review by mcmchris — CHANGES_REQUESTED** [review]
@DavidsonMSD great application note, I like the way you adapt to suggestions and templates. 
I left some comments and suggestions mainly focused on format, titles and minor stuff. Feel free to ask for more details about them. 
cc. @jcarolinares

**Review by mcmchris — APPROVED** [review]
This is ready to publish! Thanks @DavidsonMSD for doing a great job with this Application Note!

**Comment by jcarolinares** [comment]
Merging

**Pull request merged**